### PR TITLE
Expand discussion of Map Key Checker defaults

### PR DESCRIPTION
**Pull request by mernst**

### CODE REVIEW

File: docs/manual/map-key-checker.tex [hunk 1]
```diff
@@ -51,19 +51,70 @@ \section{Map key annotations\label{map-key-annotations}}
 
 \section{Default annotations\label{map-key-defaults}}
 
-The null literal is implicitly annotated with \code{@UnknownKeyFor}, because
-\code{null} might be a key for any map.
+The \code{null} literal is implicitly annotated with \code{@UnknownKeyFor}
+If \code{null} were \code{@UnknownKeyFor}, that would mean that
```
Comment (wmdietl) [inline docs/manual/map-key-checker.tex]: UnknownKeyFor should be KeyForBottom here, right?

Comment (mernst) [reply docs/manual/map-key-checker.tex]: You are right.  Thanks for the catch!

File: docs/manual/map-key-checker.tex [hunk 2]
```diff
@@ -51,19 +51,70 @@ \section{Map key annotations\label{map-key-annotations}}
 
 \section{Default annotations\label{map-key-defaults}}
 
-The null literal is implicitly annotated with \code{@UnknownKeyFor}, because
-\code{null} might be a key for any map.
+The \code{null} literal is implicitly annotated with \code{@UnknownKeyFor}
```
Comment (smillst) [inline docs/manual/map-key-checker.tex]: Missing period?

File: docs/manual/map-key-checker.tex [hunk 3]
```diff
@@ -51,19 +51,70 @@ \section{Map key annotations\label{map-key-annotations}}
...
+
+%BEGIN LATEX
+\medskip
+%END LATEX
+
+
+If lower bounds are defaulted to \code{@KeyForBottom},
+then whenever \<null> is assigned to a variable whose type is a type variable,
+programmers
+must write a \<UnknownKeyFor> annotation on either the type variable
```
Comment (smillst) [inline docs/manual/map-key-checker.tex]: an `UnknownKeyFor`?

**Comment by mernst** [comment]
@wmdietl You started a review but didn't approve the pull request.  Do you want to take a look, or not?

**Pull request merged**